plugin/metadata/README.md

```diff
@@ -8,15 +8,20 @@
 
 By enabling *metadata* any plugin that implements [metadata.Provider
 interface](https://godoc.org/github.com/coredns/coredns/plugin/metadata#Provider) will be called for
-each DNS query, at beginning of the process for that query, in order to add it's own Metadata to
+each DNS query, at beginning of the process for that query, in order to add it's own meta data to
 context.
 
-The metadata collected will be available for all plugins, via the Context parameter
-provided in the ServeDNS function. The package (code) documentation has examples on how to inspect
-and retrieve metadata a plugin might be interested in.
+The meta data collected will be available for all plugins, via the Context parameter provided in the
+ServeDNS function. The package (code) documentation has examples on how to inspect and retrieve
+metadata a plugin might be interested in.
 
-TODO: write about naming of the keys (labels).
-TODO: write about extracting and using
+The meta data is added by setting a label with a value in the context. These labels should be named
+`plugin/NAME`, where **NAME** is something descriptive. The only hard requirement the *metadata*
+plugin enforces is that the labels contains a slash. See the documentation for
+`metadata.SetValueFunc`. 
+
+The value stored is a string. The empty string signals "no meta data". See the documentation for
```

Comment:
same note as above. I would just not mention "The empty string signals "no meta data". ".